### PR TITLE
Prevent materials from generating recipes by default if they aren't solid at room temperature.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Should drastically reduce the creation and deletion of atoms during startup.

### CODE REVIEW

**Review by comma — APPROVED** [review]

File: code/modules/materials/material_recipes.dm [hunk 1]
```diff
@@ -17,8 +17,11 @@
 		. += new recipe_type(src)
 
 /decl/material/proc/generate_recipes(var/reinforce_material)
-	. = list()
 
+	if(phase_at_stp() != MAT_PHASE_SOLID)
```
Comment (out-of-phaze) [inline code/modules/materials/material_recipes.dm:21]: Why not just pass a temperature argument in? If someone's standing in subzero temperatures they might be able to craft a frozen cake batter reinforced window, for example, even if they can't at STP.

**Review by out-of-phaze — APPROVED** [review]
Seems good enough for a stopgap to prevent unnecessary recipe generation, and will need to be retouched if we redo material phases in the future anyway.

File: code/modules/materials/material_recipes.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/materials/material_recipes.dm:21]: The list is cached on first call.

Comment (MistakeNot4892) [reply code/modules/materials/material_recipes.dm:21]: To elaborate a bit - letting recipes be variable based on environmental factors would require a fairly substantial rework of how recipes are created and cached,

Comment (out-of-phaze) [reply code/modules/materials/material_recipes.dm:21]: Right. To do that we'd need to rewrite a lot anyway, so how it is in the PR is fine.

**Pull request merged**